### PR TITLE
Bugfixes n miscellanea

### PR DESCRIPTION
**Pull request by krulis-martin**

### CODE REVIEW

File: app/helpers/ExerciseConfig/Pipeline/Box/Box.php [hunk 1]
```diff
@@ -77,6 +77,17 @@ public function __construct(BoxMeta $meta) {
...
   public abstract function getCategory(): string;
 
+
+  /**
+   * Whether the box can be optimized (merged with another box with equal functionality).
+   * Boxes are optimizable by default, but specific boxes may prevent this behavior if necessary.
+   * @return bool
+   */
+  public function isOptimizable(): bool
+  {
```
Comment (Neloop) [inline app/helpers/ExerciseConfig/Pipeline/Box/Box.php]: Well, I have to say that I am kind of against having 2 formatting styles in one file :-) which clashes together... All other methods have AFAIK opening braces on the same line as signature of the function...

File: app/V1Module/presenters/ExercisesConfigPresenter.php [hunk 2]
```diff
@@ -638,20 +662,39 @@ public function actionSetTests(string $id) {
...
-    $this->configChecker->check($exercise);
-    $this->exercises->flush();
+      $exercise->updatedNow();
+      $this->exercises->flush();
 
-    $this->sendSuccessResponse($exercise->getExerciseTests()->getValues());
+      $this->configChecker->check($exercise);
+      $this->exercises->flush();
+      $this->exercises->commit();
+    }
```
Comment (Neloop) [inline app/V1Module/presenters/ExercisesConfigPresenter.php]: Unnecessary trailing newline, not the PSR-2 way in try-catch...

File: app/helpers/ExerciseConfig/Limits/ExerciseLimits.php [hunk 3]
```diff
@@ -56,6 +56,23 @@ public function removeLimits(string $testId): ExerciseLimits {
...
+  /**
+   * Remove limits for given test identification.
+   * @param string $testId
+   * @return ExerciseLimits
+   */
+  public function changeTestId(string $oldId, string $newId): ExerciseLimits {
+    $limits = $this->getLimits($oldId);
+    if ($limits) {
+      if ($this->getLimits($newId)) {
+        throw new \Exception("Serious internal error. Newly created test ID is already present in exercise limits!");
```
Comment (Neloop) [inline app/helpers/ExerciseConfig/Limits/ExerciseLimits.php]: `InternalServerException`?

File: app/config/permissions.neon [hunk 4]
```diff
@@ -514,6 +515,7 @@ permissions:
           - delete
       conditions:
           - referenceExerciseSolution.isExerciseAuthor
+          - referenceExerciseSolution.isExerciseSuperGroupAdmin
```
[the remark this answers is not in the document]
Comment (krulis-martin) [reply app/config/permissions.neon]: of cOuRse

**Pull request merged**